Firestore/core/src/firebase/firestore/remote/connectivity_monitor_apple.mm

```diff
@@ -107,9 +111,31 @@ explicit ConnectivityMonitorApple(
       LOG_DEBUG("Couldn't set reachability queue");
       return;
     }
+
+#if TARGET_OS_IOS || TARGET_OS_TV
+    this->observer_ = [[NSNotificationCenter defaultCenter]
+        addObserverForName:UIApplicationWillEnterForegroundNotification
+                    object:nil
+                     queue:[NSOperationQueue mainQueue]
+                usingBlock:^(NSNotification* note) {
+                  SCNetworkReachabilityFlags flags{};
+                  if (SCNetworkReachabilityGetFlags(reachability_, &flags)) {
+                    auto status = ToNetworkStatus(flags);
+                    if (status != NetworkStatus::Unavailable) {
+                      this->InvokeCallbacks(status);
```

Comment:
Add a comment here indicating what we're fixing. Something like:

> There may have been network changes while Firestore was in the background for which we did not get `OnReachabilityChangedCallback` notifications. If entering the foreground and we have a connection, reset the connection to ensure that RPCs don't have to wait for TCP timeouts.

Reply:
Done.